streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -377,13 +382,21 @@ void handleRevocation(final Collection<TopicPartition> revokedPartitions) {
         final Set<TopicPartition> remainingPartitions = new HashSet<>(revokedPartitions);
 
         final Map<TaskId, Map<TopicPartition, OffsetAndMetadata>> consumedOffsetsAndMetadataPerTask = new HashMap<>();
+        final Set<Task> suspended = new HashSet<>();
         for (final Task task : tasks.values()) {
             if (remainingPartitions.containsAll(task.inputPartitions())) {
                 task.prepareSuspend();
                 final Map<TopicPartition, OffsetAndMetadata> committableOffsets = task.committableOffsetsAndMetadata();
                 if (!committableOffsets.isEmpty()) {
                     consumedOffsetsAndMetadataPerTask.put(task.id(), committableOffsets);
                 }
+                suspended.add(task);
+            } else if (eosBetaEnabled && !eosUpgradeModeEnabled) {
```

Comment:
For eos-beta, we need to commit all tasks at once, hence, we prepare all non-suspended tasks for committing, too.

Reply:
`Not for this PR`: this is getting really messy here... instead of interleaving the suspending and committing logic, we should just do one loop over ALL tasks gathering their offsets, and then do another loop over the tasks from `revokedPartitions` doing a suspend.